corehq/motech/generic_inbound/utils.py

```diff
@@ -14,22 +21,79 @@ def make_url_key():
     return raw_key.removesuffix("==")
 
 
-def get_context_from_request(request):
-    try:
-        body = json.loads(request.body.decode('utf-8'))
-    except (UnicodeDecodeError, json.JSONDecodeError):
-        raise GenericInboundUserError(_("Payload must be valid JSON"))
+@attr.s(kw_only=True, frozen=True, auto_attribs=True)
+class RequestData:
```

Comment:
nit: it would be nice to name this and `ApiResponse` similarly. Maybe `Api[Request|Response]Data`?

Reply:
Agreed, I noticed that as well when writing this PR description - one of those things that's more obvious in retrospect.  Will do.